File: app/src/main/java/com/cartenz/cartenztaxcore/feature/forgot/ForgotPresenter.java


package com.cartenz.cartenztaxcore.feature.forgot;

import android.content.Context;
import android.support.annotation.Nullable;

import com.cartenz.cartenztaxcore.App;
import com.cartenz.cartenztaxcore.api.dao.SimpleStringDao;
import com.cartenz.cartenztaxcore.api.repository.ForgotPassRepository;
import com.cartenz.core.api.MySubscriber;
import com.cartenz.core.base.BaseSubscriber;

import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

final class ForgotPresenter extends BaseSubscriber implements ForgotContract.PresenterInterface {


    @Nullable
    private ForgotContract.View mForgotView;
    private Context context;

    @Override
    public void setView(Context context, ForgotContract.View view) {
        this.context = context;
        this.mForgotView = view;

    }


    @Override
    public void callForgot(String email) {
        ForgotPassRepository repo = new ForgotPassRepository(App.getApi(), email);
        addSubscription(repo.post().subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new MySubscriber<SimpleStringDao>() {
                    @Override
                    public void onComplete() {

                    }

                    @Override
                    public void onError(String message, String errorMessage) {
                        mForgotView.forgotResult(errorMessage);
                    }

                    @Override
                    public void onSuccess(SimpleStringDao string) {
                        mForgotView.forgotResult(null);
                    }

                }));

    }

    @Override
    public void dropView() {
        finishSubscriber();
        mForgotView = null;
    }

}

File: kotlinapp/src/main/java/com/cartenz/kotlinapp/feature/forgot/ForgotPresenter.kt
package com.cartenz.kotlinapp.feature.forgot

import android.content.Context
import com.cartenz.kotlin_core.CartenzBaseSubscriber
import com.cartenz.kotlin_core.api.MySubscriber
import com.cartenz.kotlinapp.api.dao.SimpleStringDao
import com.cartenz.kotlinapp.api.repository.ForgotPassRepository
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

internal class ForgotPresenter : CartenzBaseSubscriber(), ForgotContract.PresenterInterfaceCartenz {

    private var mForgotView: ForgotContract.ViewCartenz? = null
    private var context: Context? = null

    override fun setView(context: Context, view: ForgotContract.ViewCartenz) {
        this.context = context
        this.mForgotView = view
    }

    override fun callForgot(email: String) {
        val repo = ForgotPassRepository(email)
        addSubscription(repo.post().subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(object : MySubscriber<SimpleStringDao>() {
                    override fun onComplete() {

                    }

                    override fun onError(message: String?, errorMessage: String?) {
                        mForgotView!!.forgotResult(errorMessage!!)
                    }

                    override fun onSuccess(string: SimpleStringDao) {
                        mForgotView!!.forgotResult("")
                    }

                }))

    }

    override fun dropView() {
        finishSubscriber()
        mForgotView = null
    }

}
